README.md

```diff
@@ -141,15 +142,130 @@ for (index, game) in myFavoriteZeldaGames.enumerate() {
+
+```swift
+let friends = ["Chris", "Billy", "Jessica"]
+print (friends[0])
+// "Chris"
+print (friends[1])
+// "Billy"
+print (friends[2])
+// "Jessica"
+```
+
+But what if you had 100 friends? Printing values like this will become tedious quickly and if you added more friends to the list, you would have to add more `print` commands to handle them.
+
+Programming languages are very good at repetitive tasks, in fact they are better than humans, so in this lesson you will learn how to perform the same action(s) on values in an array multiple times. This is called _iterating over an array_ and introduces a new concept, a _loop_.
+
+## Loops
+
+```swift
+let friends = ["Chris", "Billy", "Jessica"]
+
+for friend in friends {
+    print("Hello \(friend)")
+}
+
+// Hello Chris
+// Hello Billy
+// Hello Jessica
+```
+
+This examples starts with a familiar array initialization, but then instead of using the `print` command for each value in the array, says that `for` each value in the `friends` array, process the instructions in-between the `{}` symbols, which are called 'curly brackets' or 'mustaches'. In this example these instructions print the current value of `friend`, or the value of the current index in the `friends` array.
+
+As `friends` is an array of type String, the type of `friend` will also be a String, and you can confirm this by typing the example above into a playground and _option + clicking_ `friend`.
+
+![Array Type Inference](array-type-inference.png)
+
+## Loops and Calculations
+
+You typically use loops to undertake more complex repetitive processing than printing the value of a variable, such as calculations.
+
+Say you have a list of numbers and want to know what the square of each one and add that value to a new list. You create one array containing the original list of numbers and a placeholder array for the list of squared numbers. Then iterate through the array containing the list of numbers, perform the necessary calculations to find it's square and add that value to the second array. Let's look at some code!
+
+```swift
+let numbers = [1, 2, 3, 4, 5]
+var squaredNumbers: [Int] = []
+
+for number in numbers {
+    let squaredNumber = number * number
+    squaredNumbers.append(squaredNumber)
+}
+
+print(squaredNumbers)
+// prints "[1, 4, 9, 16, 25]"
+```
+
+For the first `numbers` array, you use the `let` keyword as you don't want to change the values in the array. As you don't know what the values of the second `squaredNumbers` array will be yet, you initialize it first with no values. Inside the `for in` loop you generate the square of the value at the current index of the array and assign it to another unchanging variable. The next line introduces something called a _method_, which you can think of as a pre-written convenient way of performing actions on something. In this particular case `.append` adds a new value to the end of the `squaredNumbers` array.
+
```

Comment:
I would like to add something like the following to this paragraph (even though it's being asked of them in the following example to print this out which is perfect).

The first time through this `for-in` loop, number which is of type `Int`, will have a value of 1 (as 1 is the first item in the array at index 0). Using this `number` constant, 1 * 1 equals 1. We are assigning that value to the `squaredNumber` constant which is then added (or appended) to the `squaredNumbers` variable. The second time through this `for-in` loop, number will have a value of 2 (as 2 is the second item in the `numbers` array at index 1). 2 * 2 equals 4 which is assigned to the `squaredNumber` constant. This is appended to the `squaredNumbers` variable. This process will keep going until which we've iterated over all the numbers in the array. Instead of reading me go through each item, you can add various print functions to get a better sense of when things occur. See below.

^^ something like that, maybe it could be its own paragraph right after this one? What do you think?